homeassistant/components/http.py

```diff
@@ -45,6 +51,63 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
+def collect_hash(config):
+    """
+    find the password/hash information in http configuration.
+
+    Args:
+        config: configuration['http']
+
+    Returns:
+        '$<algorithm>$<salt>$<hash>'
+        or
+        None if password or hash are not configured
+    """
+    if CONF_API_HASH in config:
```

Comment:
Generally speaking, you'll want to collect all information out of the config in one location. Ideally, this function would have inputs of config_hash and config_password, or something along those lines.

Especially when config options change, it can be troublesome to have the parsing of the config data spread throughout the code.